src/components/utils/src/logger/boostlogger.cc

```diff
@@ -0,0 +1,200 @@
+
+void BoostLogger::Init() {
+  // Add formatting parameters to INI file
+  // Allows %Severity% to be used in ini config file for property Format.
+  boost::log::register_simple_formatter_factory<
+      boost::log::trivial::severity_level,
+      char>("Severity");
+  boost::log::register_simple_formatter_factory<std::string, char>("Component");
+  boost::log::register_simple_formatter_factory<boost::posix_time::ptime, char>(
+      "TimeStamp");
+  boost::log::register_simple_formatter_factory<std::thread::id, char>(
+      "ThreadId");
+  boost::log::register_simple_formatter_factory<std::string, char>("FileName");
+  boost::log::register_simple_formatter_factory<int, char>("LineNum");
+  boost::log::register_simple_formatter_factory<std::string, char>("Trace");
+
+  // Add filter parameters to INI file
+  // Allows %Severity% to be used in ini config file for property Filter.
+  boost::log::
+      register_simple_filter_factory<boost::log::trivial::severity_level, char>(
+          "Severity");
+  boost::log::register_simple_filter_factory<std::string, char>("Component");
+  boost::log::register_simple_filter_factory<std::string, char>("Trace");
+
+  std::ifstream file(filename_);
+  boost::log::settings settings = boost::log::parse_settings(file);
+
+  // Custom Settings
+
+  if (settings.has_section("Sinks")) {
+    auto& sinks = settings.property_tree().get_child("Sinks");
+    for (auto& sink : sinks) {
+      std::string sink_name = "Sinks." + sink.first;
+
+      // Disable logging for particular sinks
+      if (boost::optional<std::string> param =
+              settings[sink_name]["DisableLogging"]) {
+        bool sink_disabled = (param.get() == "true");
+        if (sink_disabled) {
+          sinks.erase(sink.first);
+        }
+      }
+    }
+  }
+
+  logging::init_from_settings(settings);
+}
+
+void BoostLogger::DeInit() {
+  logging::core::get()->remove_all_sinks();
+}
+
+logging::trivial::severity_level getBoostLogLevel(LogLevel log_level) {
+  switch (log_level) {
+    case LogLevel::TRACE_LEVEL:
+      return logging::trivial::severity_level::trace;
+    case LogLevel::DEBUG_LEVEL:
+      return logging::trivial::severity_level::debug;
+    case LogLevel::INFO_LEVEL:
+      return logging::trivial::severity_level::info;
+    case LogLevel::WARNING_LEVEL:
+      return logging::trivial::severity_level::warning;
+    case LogLevel::ERROR_LEVEL:
+      return logging::trivial::severity_level::error;
+    case LogLevel::FATAL_LEVEL:
+      return logging::trivial::severity_level::fatal;
+    default:
+      assert(false);
+  }
+}
+
+boost::posix_time::ptime BoostLogger::GetLocalPosixTime(
+    const logger::TimePoint& timestamp) {
+  auto time = std::chrono::duration_cast<std::chrono::microseconds>(
+                  timestamp.time_since_epoch())
+                  .count();
+
+  boost::posix_time::ptime time_epoch(boost::gregorian::date(1970, 1, 1));
```

Comment:
can we make this constexpr? is this not already a compile time constant in the boost lib?

Reply:
Wasn't able to find a specific constant for the epoch(January 1, 1970). Also not sure if `boost::gregorian::date` meets the constexpr Literal type requirements https://en.cppreference.com/w/cpp/named_req/LiteralType. I could make a separate constant for the `epoch_date`. Let me know what you think

Reply:
yup, i couldn't find a constant either. seems the way you're doing this is common :)